core/src/main/java/org/apache/carbondata/core/util/AbstractDataFileFooterConverter.java

```diff
@@ -153,6 +165,14 @@ private static BitSet getPresenceMeta(
       for (int i = 0; i < table_columns.size(); i++) {
         columnSchemaList.add(thriftColumnSchemaToWrapperColumnSchema(table_columns.get(i)));
       }
+      // In case of non transactional table just set columnuniqueid as columnName to support
+      // backward compatabiity. non transactional tables column uniqueid is always equal to
+      // columnname
+      if (!isTransactionalTable) {
```

Comment:
Is the columnSchema only for runtime usage?
If not, Will it be problem if use change nonTransactionalTable to transactionalTable?